src/uu/tr/src/operation.rs

```diff
@@ -0,0 +1,529 @@
+                            + b.to_digit(8).unwrap() * 8
+                            + c.to_digit(8).unwrap(),
+                    )
+                    .unwrap(),
+                ),
+            )
+        })
+    }
+
+    fn parse_backslash(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("\\")))(input).map(|(l, _)| (l, Sequence::Char('\\')))
+    }
+
+    fn parse_audible_bel(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("a")))(input).map(|(l, _)| (l, Sequence::Char('\u{0007}')))
+    }
+
+    fn parse_backspace(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("b")))(input).map(|(l, _)| (l, Sequence::Char('\u{0008}')))
+    }
+
+    fn parse_form_feed(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("f")))(input).map(|(l, _)| (l, Sequence::Char('\u{000C}')))
+    }
+
+    fn parse_newline(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("n")))(input).map(|(l, _)| (l, Sequence::Char('\u{000A}')))
+    }
+
+    fn parse_return(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("r")))(input).map(|(l, _)| (l, Sequence::Char('\u{000D}')))
+    }
+
+    fn parse_horizontal_tab(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("t")))(input).map(|(l, _)| (l, Sequence::Char('\u{0009}')))
+    }
+
+    fn parse_vertical_tab(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("\\"), tag("v")))(input).map(|(l, _)| (l, Sequence::Char('\u{000B}')))
+    }
+
+    fn parse_char_range(input: &str) -> IResult<&str, Sequence> {
+        separated_pair(take(1usize), tag("-"), take(1usize))(input).map(|(l, (a, b))| {
+            (l, {
+                let (start, end) = (
+                    u32::from(a.chars().next().unwrap()),
+                    u32::from(b.chars().next().unwrap()),
+                );
+                if (48..=90).contains(&start) && (48..=90).contains(&end) && end > start {
+                    Sequence::CharRange(
+                        (start..=end)
+                            .map(|c| std::char::from_u32(c).unwrap())
+                            .collect(),
+                    )
+                } else {
+                    Sequence::CharRange((start..=end).filter_map(std::char::from_u32).collect())
+                }
+            })
+        })
+    }
+
+    fn parse_char_star(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("["), take(1usize), tag("*"), tag("]")))(input).map(|(_, (_, _, _, _))| todo!())
+    }
+
+    fn parse_char_repeat(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("["), take(1usize), tag("*"), take_until("]"), tag("]")))(input).map(
+            |(l, (_, c, _, n, _))| {
+                (
+                    l,
+                    Sequence::CharRange(
+                        std::iter::repeat(c.chars().next().unwrap())
+                            .take(n.parse().unwrap())
```

Comment:
I think this can fail the way it's currently written, because `n` is not guaranteed to consist of digits. It would also be nice if you could add a small todo comment for the octal syntax (when `n` starts with `0`)

Reply:
Using something like `take_while1(|c: char| c.is_digit(10))` should be enough right? Although the note on octal support is warranted.

Reply:
Yes, I'd go for `digit1` (which is essentially the same thing)